app.js

```diff
@@ -20,19 +10,46 @@ You are running the app using production config (Firestore Database, Map API key
 
 If you are developing the app locally, please use "npm run dev" to start the app.
 `;
+
+// Load API keys.
+if (process.env.NODE_ENV !== 'production') {
+  require('dotenv').config();
+} else {
+  console.log(PROD_WARNING_MESSAGE);
+}
+
+// Module dependencies.
+var express = require('express');
+var http = require('http');
+var path = require('path');
+var handlebars = require('express-handlebars')
+var passport = require('passport');
+var cookieSession = require('cookie-session');
+require('./passport-auth');
+
+var app = express();
+
 // Environments configs. 
 app.set('port', process.env.PORT || 3000);
 app.set('views', path.join(__dirname, 'views'));
 app.engine('handlebars', handlebars());
 app.set('view engine', 'handlebars');
 app.use(express.urlencoded());
 app.use('/static', express.static('public'));
+app.use(cookieSession({
+  name: 'auth-session',
+  keys: ['key1', 'key2']
+}));
+app.use(passport.initialize());
+app.use(passport.session());
 
-// Load API keys.
-if (process.env.NODE_ENV !== 'production') {
-  require('dotenv').config();
-} else {
-  console.log(PROD_WARNING_MESSAGE);
+// Authentication middleware
+const isLoggedIn = (req, res, next) => {
+  if (req.user) {
+    next();
+  } else {
+    res.sendStatus(401);
```

Comment:
Is is possible to redirect users to the login page if they try to access a page while not logged in? I'm not sure if that would mess up how you use isLoggedIn in other places.

Reply:
+1, this is model most google pages follow. We could do this in a follow up though.

Reply:
I'll make another PR for this.